### PR TITLE
Feature/dmp bam pairing

### PR DESCRIPTION
**Pull request by allanbolipata**
Adds dmp bam pairing.

I changed default pooled normal file group [here](https://github.com/mskcc/beagle/compare/develop...feature/dmp_bam_pairing?expand=1#diff-202d73738f891d89584dcdb6b06414c8R282
) so it coincides with the uuid assigned on production 

Changed it [in fixtures json](https://github.com/mskcc/beagle/compare/develop...feature/dmp_bam_pairing?expand=1#diff-e751a8f21ccf77b468e66baf67af7935R26), too

### CODE REVIEW

**Comment by stevekm** [comment]
the code looks nice but is there a test case for `get_dmp_normal` and `build_dmp_query`? I cannot easily tell what their input/output and other dependencies (e.g. db entries) look like

**Comment by stevekm** [comment]
`get_pooled_normals` does not appear to work properly when the pooled normals have metadata (`FileMetadata.metadata`) that looks like this:

```
"recipe": "IMPACT468",
'baitSet': 'IMPACT468_BAITS',
```

That is causing the other tests to fail, because it is not picking up the simulated pooled normals. 

Is this a bug or an intended behavior?

**Comment by allanbolipata** [comment]
> `get_pooled_normals` does not appear to work properly when the pooled normals have metadata (`FileMetadata.metadata`) that looks like this:
> 
> ```
> "recipe": "IMPACT468",
> 'baitSet': 'IMPACT468_BAITS',
> ```
> 
> That is causing the other tests to fail, because it is not picking up the simulated pooled normals.
> 
> Is this a bug or an intended behavior?

I don't know how you are testing this, but it works for `07008_DA`, which has a recipe of IMPACT468 and a baitSet of IMPACT468_BAITS. It pairs to pooled normals http://silo:4001/v0/run/api/8ae810b6-290a-4632-ab91-895404ed2d7c/

Are we sure it's testing properly?

File: runner/operator/roslin_operator/roslin_operator.py [hunk 1]
```diff
@@ -71,6 +71,10 @@ def get_jobs(self):
                 sample_mapping += "\t".join(
                     [normal_sample_name, FileProcessor.parse_path_from_uri(p['location'])]) + "\n"
 
+            for p in job['pair'][1]['bam']:
+                sample_mapping += "\t".join(
+                    [normal_sample_name, FileProcessor.parse_path_from_uri(p['location'])]) + "\n"
```
Comment (sivkovic) [inline runner/operator/roslin_operator/roslin_operator.py:76]: Now I see we are using it here. We should try doing this everywhere we have path manipulation.

File: runner/operator/roslin_operator/construct_roslin_pair.py [hunk 2]
```diff
@@ -50,6 +50,8 @@ def format_sample(data):
         sample[r1].append({'class': 'File', 'location': 'juno://' + i})
     for i in data['R2']:
         sample[r2].append({'class': 'File', 'location': 'juno://' + i})
+    for i in data['bam']:
+        sample['bam'].append({'class': 'File', 'location': 'juno://' + i})
```
[the remark this answers is not in the document]
Comment (allanbolipata) [reply runner/operator/roslin_operator/construct_roslin_pair.py:54]: We'll add it in a future update

**Pull request merged**